js_modules/dagster-ui/packages/ui-core/src/graph/asyncGraphLayout.ts

```diff
@@ -139,7 +173,17 @@ export function useAssetLayout(graphData: GraphData) {
 
     async function runAsyncLayout() {
       dispatch({type: 'loading'});
-      const layout = await asyncGetFullAssetLayout(graphData, opts);
+      let layout;
+      if (localStorageKey) {
+        layout = await asyncGetFullAssetLayoutLocalStorage(
+          localStorageKey,
+          fullDataCacheKey ?? cacheKey,
+          graphData,
+          opts,
+        );
+      } else {
+        layout = await asyncGetFullAssetLayout(graphData, opts);
```

Comment:
Is it just the asset lineage page graph that falls into this else case? It might be better to "cache large graphs only" or "cache if it takes >500ms to build" or something like that instead of making it an option whether you want storage.